File: Python DataStruc/Euler's Problems/largestprimefactor.c


#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int isprime(long int num);

int isprime(long int num)
{
int i;
for(i=2; i<=num; i++)
{
	if(num%i==0)
	{
		return 1;
	}
}

return 0;
}



int main()
{
long int num = 600851475143;
long int i;
int array[10000];

for(i=2; i<=num; i++)
{
	if(isprime(i)==0)
	{
		if(num%i==0)
		{
			printf("%ld",i);
		}	
	}
}


return 0;
}

File: Python DataStruc/Euler's Problems/largestprimefactor.py
##Python is so fucking slow

def isprime(num):
	for i in range(2, int(num**0.5)+1):
		if(num % i == 0):
			return False
	return True

def main():
	num = 600851475143
	list_prime = []
	for i in range(2,num+1):
		if(isprime(i)==True):
			if(num % i == 0):
				list_prime.append(i)
			else:
				continue

	print(max(list_prime))
		
	
main()
